forc-plugins/forc-client/src/op/call/transaction_trace.rs

```diff
@@ -0,0 +1,1176 @@
+                writeln!(
+                    f,
+                    "{}{} [{}] {}",
+                    indent,
+                    prefix,
+                    gas,
+                    Color::Green.paint(format!("0x{}", to))
+                )?;
+                // Format children
+                for child in &self.children {
+                    child.fmt_with_depth(f, depth + 1)?;
+                }
+                Ok(())
+            }
+            Receipt::Return { val, .. } => {
+                writeln!(f, "{}{} ← [Return] val: {}", indent, return_prefix, val)
+            }
+            Receipt::ReturnData { data, .. } => {
+                let data_str = match data {
+                    Some(data) if !data.is_empty() => format!("0x{}", hex::encode(data)),
+                    _ => "()".to_string(),
+                };
+                writeln!(
+                    f,
+                    "{}{} ← {}",
+                    indent,
+                    return_prefix,
+                    Color::BrightCyan.paint(data_str),
+                )
+            }
+            Receipt::Revert { .. } => {
+                writeln!(
+                    f,
+                    "{}{} ← {}",
+                    indent,
+                    return_prefix,
+                    Color::Red.paint("[Revert]")
+                )
+            }
+            Receipt::Panic { reason, .. } => {
+                let reason_str = format!("{:?}", reason);
+                writeln!(
+                    f,
+                    "{}{} ← {} {}",
+                    indent,
+                    return_prefix,
+                    Color::Red.paint("[Panic]"),
+                    Color::Red.paint(reason_str)
+                )
+            }
+            Receipt::Log { rb, .. } => {
+                writeln!(f, "{}{} [Log] rb: 0x{:x}", indent, prefix, rb)
+            }
+            Receipt::LogData { id, rb, data, .. } => {
+                let data_str = match data {
+                    Some(data) => {
+                        let hex_str = format!("0x{}", hex::encode(data));
+                        match self.abis.and_then(|abis| abis.get(id)) {
+                            Some(abi) => {
+                                let program_abi = sway_core::asm_generation::ProgramABI::Fuel(
+                                    abi.program.clone(),
+                                );
+                                forc_util::tx_utils::decode_log_data(
+                                    &rb.to_string(),
+                                    data,
+                                    &program_abi,
+                                )
+                                .ok()
+                                .map(|decoded| decoded.value)
+                                .unwrap_or(hex_str)
+                            }
+                            None => hex_str,
+                        }
+                    }
+                    None => "".to_string(),
```

Comment:
Is this state reachable?

Reply:
I have not been able to verify this - i.e. logging data in sway - with no data.
But since it's an optional type - we need to explicitly handle this case (so in theory it should be possible)